src/passes/StackIR.cpp

```diff
@@ -292,6 +356,16 @@ class StackIROptimizer {
 
   bool isControlFlow(StackInst* inst) { return inst->op != StackInst::Basic; }
 
+  // The beginning of an instruction: either the beginning of a control flow,
+  // or a basic instruction in which case this is also the end.
+  bool isBegin(StackInst* inst) {
```

Comment:
I would name this something like `isBeginOrBasic`, since `isBegin` suggests to me that it is exclusive of basic instructions. Same with`isEnd`.